mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -129,13 +133,24 @@ bool MemRefDependenceGraph::init() {
       for (auto *opInst : collector.loadOpInsts) {
         node.loads.push_back(opInst);
         auto memref = cast<AffineReadOpInterface>(opInst).getMemRef();
+        if (memrefAccesses.find(memref) == memrefAccesses.end()) {
+          for (auto *n : memoryEffectNodes)
+            memrefAccesses[memref].insert(n->id);
+        }
         memrefAccesses[memref].insert(node.id);
       }
       for (auto *opInst : collector.storeOpInsts) {
         node.stores.push_back(opInst);
         auto memref = cast<AffineWriteOpInterface>(opInst).getMemRef();
+        if (memrefAccesses.find(memref) == memrefAccesses.end()) {
+          for (auto *n : memoryEffectNodes)
+            memrefAccesses[memref].insert(n->id);
+        }
         memrefAccesses[memref].insert(node.id);
       }
+      node.memEffects = collector.memoryEffectOpInsts;
```

Comment:
You are combining ops with all memory effects but you need those with write effects below. See below.

Reply:
Can you respond to this comment?

Reply:
Currently, we are considering side-effects as unknown entities that could be potentially read or write ops. However, if we consider side-effects with only write effect. we might lose out on potential WAR dependency in the MDGs.

See comment response https://github.com/mcl-csa/llvm-project-public/pull/13#discussion_r1470726826